### DISCUSSION TITLE
JEDI-2P & ULoVE:  not exactly calcium imaging nor electrophysiology

### DISCUSSION THREAD

**Discussion by phanhuynh** [NeurodataWithoutBorders/helpdesk#67] [General]
Hello all, I'm working with data that's in between calcium imaging and intracellular electrophysiology because it's recorded on a 2photon microscope, but the data itself resembles voltage recording.  The technique is called ultrafast local volume excitation (ULoVE):  

https://www.sciencedirect.com/science/article/pii/S0092867422009163?ref=pdf_download&fr=RR-2&rr=8476356bfdb771df

Has anyone here worked with this, and has advice for how to create an NWB for it?
[1-s2.0-S0092867422009163-main (1).pdf](https://github.com/NeurodataWithoutBorders/helpdesk/files/13975619/1-s2.0-S0092867422009163-main.1.pdf)

**Comment by CodyCBakerPhD** [comment]
I'll defer to the expert, @alessandratrapani, but if it is acquired with the TwoPhoton microscope, then you would use the [TwoPhoton microscopy](https://pynwb.readthedocs.io/en/stable/tutorials/domain/ophys.html#sphx-glr-tutorials-domain-ophys-py) representation (I know the tutorial says 'Calcium Imaging', but the only thing specific to calcium recordings there is the indicator; as far as I know, your voltage indicator is the only major difference in the setup right?)

Of course if there are any obvious issues that you spot with that representation, raise them as an issue on [ndx-microscopy](https://github.com/catalystneuro/ndx-microscopy), the upcoming extension that plans to eventually replace those core data types

**Reply by phanhuynh** [reply]
Thank you, Cody!

**Comment by phanhuynh** [comment]
@alessandratrapani, hello!  I was wondering if you had any additional thoughts on this?  

Thank you :)

**Reply by alessandratrapani** [reply]
Hi @phanhuynh, sorry for taking so long to answer; I was unfamiliar with the technique and had to research a bit before getting back to you.
I do agree with Cody.
The main difference seems to reside in the type of excitation you deliver to the tissue to obtain the fluorescent signal. With this technique, you can simultaneously excite a limited number of fixed focal points in a defined area (holographic image), correct?
The only property that is required to describe the type of excitation is the “excitation_lambda” that you may find in the ImagingPlane object.
And, of course, as Cody pointed out, the indicator, in your case a GEVI (JEDI-2P). You may provide this information also in the ImagingPlane object.
As far as I understood, the raw fluorescence signal that you obtain is volumetric imaging data of a given dimension in pixel [width,height,depth] and imaging rate (at the kHz order in this case), that you can acquire using different channels, which can be described by `OpticalChannel` property of the `ImagingPlane`. 
Also, you would still express the Voltage traces as DF/F signal, which can be stored in the `DfOverF` container as `RoiResponseSeries`.
In general, I would recommend to add a detailed description of this technique in either in the Device stored in the ImagingPlane object, or in the `experiment_description` property of the NWBFile .

**Reply by phanhuynh** [reply]
Allessandra, I am so grateful for how much time you put into your response!
